src/main/java/com/unorganized/dao/TaskListDAO.java

```diff
@@ -6,21 +6,21 @@
 import java.io.IOException;
 import java.io.ObjectInputStream;
 import java.io.ObjectOutputStream;
-import java.io.Serializable;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 
 import com.unorganized.dto.TaskDTO;
 
-public class TaskListDAO implements Serializable
+public class TaskListDAO implements ITaskListDAO
```

Comment:
1. Serializable should be used for entity/DTO objects, not DAOs.
2. DAO implementations should implement the interface to ensure polymorphism.